### PR TITLE
Reset sharedialog values

### PR DESCRIPTION
**Pull request by rullzer**
By not resetting the sharedialog values stuff breaks! See #15368

This resets the values to the correct state so the code works as expected again.

No unit tests yet as the IE8 check at https://github.com/owncloud/core/blob/master/core/js/share.js#L1012 breaks my unit tests. I already called in the help of @PVince81 about that.

Other than the unit test this can be tested already

### CODE REVIEW

File: core/js/share.js [hunk 1]
```diff
@@ -1009,7 +1013,7 @@ $(document).ready(function() {
 			} else {
 				$('#linkPass').slideToggle(OC.menuSpeed);
 				// TODO drop with IE8 drop
-				if(html.hasClass('ie8')) {
+				if($('html').hasClass('ie8')) {
```
Comment (rullzer) [inline core/js/share.js]: Can we safely do this? Else the unit tests won't work here. And Firefox is complaining about it as well...

**Comment by rullzer** [comment]
We now have unit tests!
Milestone to 8.1 as it would be good to have it in the current release as it can lead to bad UX.

CC: @PVince81 @MorrisJobke @DeepDiver1975 @LukasReschke

File: core/js/share.js (hunk 1 above)
Comment (MorrisJobke) [inline core/js/share.js]: Yes. This is actually the way it should be.

Comment (DeepDiver1975) [inline core/js/share.js]: wasn't that fixed recently?

Comment (rullzer) [inline core/js/share.js]: I just rebased. And no conflict. So maybe we had this is several places.

**Comment by PVince81** [comment]
Tested before and after the fix: works :+1:

**Comment by PVince81** [comment]
needs a second reviewer @nickvergessen

**Comment by nickvergessen** [comment]
Seems to cause weir behaviour for me:

1. Open the dialog
2. Share with Link
3. Check password box
4. Set password
5. Uncheck the link option
6. Share with link again

### Expected

The password field is either checked or collapsed

### Actual

The box is checked, but the password field is collapsed, so in order to enter a password you need to disable the password option… wait woot? :stuck_out_tongue_winking_eye: :trollface:

**Comment by rullzer** [comment]
@nickvergessen seems this is not introduced by this code. But is also a bug in master. Never the less should be fixed :P

O to test this you need to enforce password protection ;) I should have mentioned that.

**Comment by nickvergessen** [comment]
Okay, raised #15812

**Comment by rullzer** [comment]
@nickvergessen could you still review this? Enforce passwords and follow what is described in #15368

**Comment by MorrisJobke** [comment]
Beside the problem that @nickvergessen found this works and properly resets the fields :+1:

**Pull request merged**